### PR TITLE
IBX-215: Added semantic configuration for password related views

### PR DESCRIPTION
**Pull request by ViniTou**
| Question      | Answer
| ------------- | ---
| Tickets       |  [IBX-215](https://jira.ez.no/browse/IBX-215) 
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

For reasons unknown, some views were lacking proper semantic configuration and were only configurable thru whole parameter name. 

#### Checklist:
- [x] Implement tests
- [x] Coding standards (`$ composer fix-cs`)

### CODE REVIEW

File: src/bundle/DependencyInjection/Configuration/Parser/ResetPassword.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformUserBundle\DependencyInjection\Configuration\Parser;
+
+use eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\AbstractParser;
+use eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\SiteAccessAware\ContextualizerInterface;
+use Symfony\Component\Config\Definition\Builder\NodeBuilder;
+
+class ResetPassword extends AbstractParser
```
Comment (adamwojs) [inline src/bundle/DependencyInjection/Configuration/Parser/ResetPassword.php:15]:
```suggestion
final class ResetPassword extends AbstractParser
```

File: src/bundle/DependencyInjection/Configuration/Parser/ForgotPassword.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformUserBundle\DependencyInjection\Configuration\Parser;
+
+use eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\AbstractParser;
+use eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\SiteAccessAware\ContextualizerInterface;
+use Symfony\Component\Config\Definition\Builder\NodeBuilder;
+
+class ForgotPassword extends AbstractParser
```
Comment (adamwojs) [inline src/bundle/DependencyInjection/Configuration/Parser/ForgotPassword.php:15]:
```suggestion
final class ForgotPassword extends AbstractParser
```

File: src/bundle/DependencyInjection/Configuration/Parser/ChangePassword.php [hunk 3]
```diff
@@ -31,8 +31,8 @@ public function addSemanticConfig(NodeBuilder $nodeBuilder)
                             ->scalarNode('form')
                                 ->info('Template to use for change password form rendering.')
                             ->end()
-                            ->scalarNode('confirmation')
-                                ->info('Template to use for change password confirmation rendering.')
+                            ->scalarNode('success')
```
Comment (adamwojs) [inline src/bundle/DependencyInjection/Configuration/Parser/ChangePassword.php:34]: Is it intentional BC break?

**Review by adamwojs — COMMENTED** [review]
Final review after providing tests coverage, but at the current PR state:

File: src/bundle/DependencyInjection/Configuration/Parser/ChangePassword.php (hunk 3 above)
Comment (ViniTou) [reply src/bundle/DependencyInjection/Configuration/Parser/ChangePassword.php:34]: I would say this is not BC break as `user_change_password.templates.confirmation` parameter was never used ;)

**Review by Steveb-p — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

**Review by adamwojs — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved Ibexa-DXP 3.3.2-dev with diff.

**Comment by lserwatka** [comment]
Could you merge it up?

**Pull request merged**